### ISSUE TITLE
Short Link of a Page does nothing when testing

### ISSUE THREAD

**Issue by bnorw24** [FlutterFlow/flutterflow-issues#1527]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Using "Generate Short Link" in "Generate Current Page Link" action produces the short link correctly, however, attempting to navigate to the short link does not navigate but long link does. 

For example: 
This shortened URL **does not work**: https://artden.page.link/381b

But the long version does: https://artden.page.link/?sd=Commission%20Share%20Link&st=Commission%20Share%20Link&apn=com.artden.app&ibi=com.artden.app&isi=1659234106&link=https%3A%2F%2Fartden.page.link%2Fcommission%2F6IdiT2sPuptwfS6aALZG%2F2c0ic1POfUSRMOlVARtmFGDY95i2

Seems the link is trying to navigate to 381b based on local logging so short link is not getting parsed BEFORE launching

### Expected Behavior

Short link would navigate to the correct page with the parameters, just like the long link

### Steps to Reproduce

1. Turn on Firebase Dynamic Links (Following steps [here](https://docs.flutterflow.io/advanced-functionality/deep-and-dynamic-linking#1.-set-url-scheme))
2. Create a page with a "Generate Current Page Link" with "Generate Short Link"
3. Note: I have page params to nav to a specific screen: commission/:commissionRefId/:artistId. Params may be needed to replicate.
4. Generate the short link and attempt to launch
5. Notice: App is launched but you are stuck on the dashboard

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFTl8jc2It2sdtF+KrUKsIxmigQM0MbWoIzlO1Ca0kdB4joPLJ+eMjRaltCZPXiYXlhHFf8oHkepdrNi+HpDvk4Jk+BGIh+1KpcTgrvUXq8RLqOOqqoenNQM+xQFVSY056RuAp4D7JpWCFl1F2uFdm/UgPbCZ+/Zwh9f6PHaOI=

### Context

The short link is much nicer to use and share - the long link is ugly and I'm not sure my users will want to use it. 

### Visual documentation

Logs:
[GoRouter] No initial matches: /381b
[GoRouter] Using MaterialApp configuration
[GoRouter] pushing /commission/6IdiT2sPuptwfS6aALZG/2c0ic1POfUSRMOlVARtmFGDY95i2

When the last log occurs, nothing happens and I cannot navigate to ANY other screen (like the app navigation is broken now)

### Environment

```markdown
- FlutterFlow version: v4.0 - Flutter 3.10.4
- Platform: Android
- Browser name and version: Chrome 116.0.5845.188
- Operating system and version affected: Android/Samsung Galaxy s9
```

**Comment by bnorw24** [comment]
👋🏻 Let me know if I can provide any additional details to make it easier for you all!

**Comment by katherineqian** [comment]
Hi @bnorw24, just to let you know I'm investigating this bug. It seems like dynamic query parameters are an issue with Firebase Dynamic links in general, especially with shortened ones  (see https://github.com/firebase/flutterfire/discussions/5579) -- so I will have to see if we can enable this. It may have to be a feature request, but I'll keep you posted.

**Comment by bnorw24** [comment]
Hello @katherineqian ! I wanted to check in on this as its been a month and deeplinking into apps is still broken so my users cannot share profiles or posts :( Can you please advise on at least a workaround for me so I can get this up and running?

**Comment by katherineqian** [comment]
Hey @bnorw24, I'm sorry about the delay. Unfortunately I haven't gotten the chance to investigate this. To clarify, would using the original long version of the deep link work as a workaround? Or is there something preventing the long link being used?

**Comment by bnorw24** [comment]
The long version of the URLs no longer works as well. Causes the same issue of freezing the app

**Comment by bnorw24** [comment]
cc @katherineqian

**Comment by bnorw24** [comment]
I cannot use regular page links either, I need deeplink functionality. Let me know if there's some custom solution I should add to my flutterflow project to get around the issue while you continue to investigate

**Comment by bnorw24** [comment]
iOS app store first open causes navigation to break in a similar way (e.g. if I force quit the app and press "Open" in ios app store on my app, i cannot navigate to any new screens), I wonder if these are related (if not let me know and I'll open a new bug ticket..) @katherineqian

**Comment by bnorw24** [comment]
@katherineqian here are videos of this

This is it working as expected, I can tap on the commission, tap edit etc:

https://github.com/FlutterFlow/flutterflow-issues/assets/47465401/eb1cc328-289e-4e44-a2ad-c9e6064e855f


This is it not working as expected, on fresh install, the first launch fails to allow you to tap on a commission, tap edit etc.

https://github.com/FlutterFlow/flutterflow-issues/assets/47465401/30fe44a4-0468-4ea9-b10b-7a4923e09ab6


This behavior is similar to the deeplink issue, e.g. on deeplink it opens app but does not launch the screen and does not allow you go to anywhere. Here is a deeplink into the commission which should launch the commission (using long URL) and I try to tap on the commission on dashboard:

https://github.com/FlutterFlow/flutterflow-issues/assets/47465401/0dcf554c-d3f6-4e09-b3bd-a8892da012b2


I cannot replicate with the long links though when I've already launched the app at least once. Seems to only happen if its a fresh install.

**Comment by katherineqian** [comment]
Hi @bnorw24, I've been investigating your report today and wanted to get back to you with some info and suggestions to hopefully unblock you as I know this is likely very frustrating for you and I hope you can get your app working ASAP too. I've tried to test this by running your project on emulator and it's quite difficult to debug because I do not have access to your Firebase project and can't verify how things have been set up and if there are any issues on that front. **And,** I searched our issue tracker and found that https://github.com/FlutterFlow/flutterflow-issues/issues/1386 looks like it could very likely be the same root problem as what you are experiencing and a lot of other users are encountering the same problem on iOS. Personally what I would do is use the Branch.io workaround that a user suggested on #1386 -- it looks like it's free for <10k MAUs, though I'm not sure if that fits your use case. For next steps on my end, I'll check in on what's going on with the wider iOS Dynamic Links issue internally.

Firebase Dynamic Links itself is being shut down, by the way - another reason why I would recommend moving off of it. 
> Deprecated: Firebase Dynamic Links is deprecated and should not be adopted in projects that don't already use it. The service will shut down on August 25, 2025. See the [Dynamic Links Deprecation FAQ](https://firebase.google.com/support/dynamic-links-faq) for more information.

Another workaround that I would personally consider to unblock your users ASAP would be to forgo dynamic links and publish your app on web, which should be quite easy in FF. Then, you can just use links to the web app for all shared links (profiles, posts) and if the user is on mobile, show a button to open in-app and use a deeplink to take the user to the same post or profile in-app -- you could even include a banner linking to the app store page encouraging people to download the app first. This is a flow I've seen before in other apps. 

Sorry for the not-so-good news. I hope this can at least be of some help while we look for a fix.

**Comment by bnorw24** [comment]
Hey @hariprasadms or @katherineqian ! Curious if there is any update on this or when FlutterFlow might be replacing Firebase Dynamic Links since they are going away? The recommended solution requires me to manually export the code and add the new library, which I'm concerned will break the FlutterFlow integration and force me to code manually

Let me know! Thanks

**Comment by bnorw24** [comment]
Update to this - seems to be Android specific

**Comment by Alezanello** [comment]
Hello,

I apologize for the late response. Does this issue still persist, or was it fixed in the latest FlutterFlow updates?

**Comment by Alezanello** [comment]
Closing this issue since we haven't heard back from you. If you are still facing the problem please submit a new issue. Thank you!
